src/ray/gcs/pubsub/gcs_pub_sub.cc

```diff
@@ -297,15 +360,44 @@ Status GcsSubscriber::SubscribeActor(
 }
 
 Status GcsSubscriber::UnsubscribeActor(const ActorID &id) {
+  if (subscriber_ != nullptr) {
+    subscriber_->Unsubscribe(rpc::ChannelType::GCS_ACTOR_CHANNEL, gcs_address_,
+                             id.Binary());
+    return Status::OK();
+  }
   return pubsub_->Unsubscribe(ACTOR_CHANNEL, id.Hex());
 }
 
 bool GcsSubscriber::IsActorUnsubscribed(const ActorID &id) {
+  if (subscriber_ != nullptr) {
+    return !subscriber_->IsSubscribed(rpc::ChannelType::GCS_ACTOR_CHANNEL, gcs_address_,
+                                      id.Binary());
+  }
   return pubsub_->IsUnsubscribed(ACTOR_CHANNEL, id.Hex());
 }
 
 Status GcsSubscriber::SubscribeAllNodeInfo(
     const ItemCallback<rpc::GcsNodeInfo> &subscribe, const StatusCallback &done) {
+  if (subscriber_ != nullptr) {
+    auto subscribe_item_callback = [subscribe](const rpc::PubMessage &msg) {
+      RAY_CHECK(msg.channel_type() == rpc::ChannelType::GCS_NODE_INFO_CHANNEL);
+      subscribe(msg.node_info_message());
+    };
+    auto subscription_failure_callback = [](const std::string &, const Status &status) {
+      RAY_LOG(WARNING) << "Subscription to NodeInfo channel failed: "
+                       << status.ToString();
+    };
+    if (subscriber_->SubscribeAll(
+            std::make_unique<rpc::SubMessage>(), rpc::ChannelType::GCS_NODE_INFO_CHANNEL,
+            gcs_address_, [done](Status status) { done(status); },
+            std::move(subscribe_item_callback),
+            std::move(subscription_failure_callback))) {
+      return Status::OK();
+    }
+    return Status::ObjectExists(
+        "NodeInfo channel already subscribed. Please unsubscribe first if it needs to be "
+        "resubscribed.");
+  }
```

Comment:
do you mind adding else to make it explicit?

Reply:
Same as above.